File: rover_flutter/lib/src/api/speech_models.dart
class RenderSpeechRequest {
  const RenderSpeechRequest({
    required this.text,
    required this.purpose,
    this.locale = 'en-US',
    this.walkSessionId,
    this.stopId,
    this.idempotencyKey,
    this.cacheEligible = false,
    this.cacheExpiresUtc,
    this.storyId,
    this.variantId,
  });

  final String text;
  final String purpose;
  final String locale;
  final String? walkSessionId;
  final String? stopId;
  final String? idempotencyKey;
  final bool cacheEligible;
  final DateTime? cacheExpiresUtc;
  final String? storyId;
  final String? variantId;

  Map<String, Object?> toJson() {
    return {
      'text': text,
      'purpose': purpose,
      'locale': locale,
      'walkSessionId': walkSessionId,
      'stopId': stopId,
      'idempotencyKey': idempotencyKey,
      'cacheEligible': cacheEligible,
      'cacheExpiresUtc': cacheExpiresUtc?.toUtc().toIso8601String(),
      'storyId': storyId,
      'variantId': variantId,
    };
  }
}

class RenderedSpeechAudio {
  const RenderedSpeechAudio({
    required this.bytes,
    required this.contentType,
    required this.provider,
    required this.cacheStatus,
    required this.usedFallback,
    this.fallbackReason,
  });

  final List<int> bytes;
  final String contentType;
  final String provider;
  final String cacheStatus;
  final bool usedFallback;
  final String? fallbackReason;
}

File: rover_middleware/Rover.Application/Speech/SpeechModels.cs
namespace Rover.Application.Speech;

public enum SpeechPurpose
{
    StopNarration,
    AskRoverAnswer,
    WalkIntroduction,
    WalkRecap,
    DiscoveryDescription
}

public sealed record RenderSpeechCommand(
    Guid AccountId,
    string Text,
    SpeechPurpose Purpose,
    string? Locale,
    string? WalkSessionId,
    string? StopId,
    string? IdempotencyKey)
{
    public bool CacheEligible { get; init; }
    public DateTimeOffset? CacheExpiresUtc { get; init; }
    public string? StoryId { get; init; }
    public string? VariantId { get; init; }
}

public sealed record RenderedSpeech(
    byte[] Audio,
    string ContentType,
    string Provider,
    bool CacheHit,
    bool UsedFallback,
    string CorrelationId,
    string? FallbackReason = null);

public sealed record SpeechUsageEvent(
    Guid AccountId,
    SpeechPurpose Purpose,
    int CharacterCount,
    bool CacheHit,
    bool Success,
    long ProviderLatencyMs,
    DateTimeOffset CreatedAtUtc);
